### ISSUE TITLE
No intellisense, can't set breakpoints

### ISSUE THREAD

**Issue by nrandell** [Microsoft/visualfsharp#5260]
I've got a netcoreapp2.1 solution with many projects. Most projects are C#, but one is F#. The F# project references a couple of C# projects.

I'm using VS 2017 15.7.4 with the nightly builds (15.7.20180627.1 at the moment)

Building and running works fine - in fact it's perfect!

However, when there is a problem in my F# code, it's a nightmare. VS 15.7.3 was much better.

At the moment, I can't do the following

1 - get any intellisense in the F# project - I need to use VS code to if I need to do significant edits.
2 - set breakpoints. System.Diagnostics.Debugger.Break() works, but I can't set a breakpoint using visual studio.
3 - See the squiggles where errors occur. I can see them in the solution explorer, but when I open the file, they disappear. 

I know this is a vague issue, but I'm more than happy to help. Unfortunately I can't provide the code as it contains customer specific details in and the test data is customer specific. But if someone could provide me a pointer to help diagnose, I'm more than happy to try and help.

**Comment by cartermp** [comment]
@nrandell Two questions:

1. Are you using Paket? If so, you'll need to `paket install` or `paket update` before opening up in VS.
2. If you back off of the nightly feed, do things work?

For (2), we don't guarantee that nightlies will be a good experience, since it's just a built from master of our codebase. Since we're deep in the process of shutting down and assembling the payload for 15.8, things that go into master may not work for 15.7.x. This sort of thing happens around the time of release, and the normal recourse is to just stop consuming nightlies until the next major VS version is shipped.

**Comment by nrandell** [comment]
Thanks @cartermp - I'm not using paket, pure nuget through the csproj and fsproj

I did try uninstalling the nightly yesterday and noticed no real change. 

I will try the latest 15.8 preview and update this issue.

**Comment by forki** [comment]
@cartermp why do you guys always think it's paket related when there is a intellisense issue? It usually turns out to be unrelated ;-)

@nrandell it usually helps to upload a zip with a repro.

**Comment by nrandell** [comment]
I can't really upload a zip of the original code due to customers algorithms. However I will see if I can make a smaller repro.

First things though, I'm going to try 15.8 after a reboot ...

**Comment by cartermp** [comment]
Looking at the [commit log for 15.7.4](https://github.com/Microsoft/visualfsharp/commits/Visual-Studio-2017-Version-15.7.4), the only thing I can see as relevant is a performance improvement for .NET SDK projects, though I think we would have heard it loud and clear if that generally broke .NET SDK-style projects somehow. The rest are bug fixes.

Another possible issue to assess - can you try closing all C# documents, close the solution, then re-open? If IntelliSense and the like light up, then this is a duplicate of #5170, which doesn't appear to indicate a regression from 15.7.3 to 15.7.4.

If that workaround doesn't do anything, would you be comfortable using Report a Problem in VS and using a trace? All data generated and sent to VS is GDPR-compliant and also under NDA; i.e., we won't be speaking about it on GitHub unless it involves a technical write-up of the bug from the F# tools perspective.

**Comment by nrandell** [comment]
@cartermp - happy to report a problem using VS and generating a trace. I'll do that later today.
In the meantime, I've disabled all extensions and things are slowly starting to work.

Some interesting things I've seen in the F# Language Service output

```[F#][INFO 06:55:31 AM] Rechecking code due to buffer edit!
[F#][ERROR 06:55:32 AM] Line Lens startup failed with: System.ArgumentException: The option value was None
Parameter name: option
   at Microsoft.FSharp.Core.OptionModule.GetValue[T](FSharpOption`1 option)
   at System.Lazy`1.CreateValue()
   at System.Lazy`1.LazyInitValue()
   at System.Lazy`1.get_Value()
   at <StartupCode$FSharp-Editor>.$FSharpCodeLensService.clo@136-175.Invoke(Unit _arg9)
   at Microsoft.FSharp.Control.AsyncPrimitives.CallThenInvokeNoHijackCheck[a,b](AsyncActivation`1 ctxt, FSharpFunc`2 userCode, b result1)
   at <StartupCode$FSharp-Editor>.$FSharpCodeLensService.clo@135-173.Invoke(AsyncActivation`1 ctxt)
   at Microsoft.FSharp.Control.Trampoline.Execute(FSharpFunc`2 firstAction)
```
But I no longer have the messages repeated and it is now working.

I can set breakpoints and even navigate to definitions in C# code (that is really nice)

I will update this comment with the extension that breaks it!

**Comment by vasily-kirichenko** [comment]
@nrandell I recommend to turn off "lenses" and never turn on it again.

**Comment by nrandell** [comment]
@cartermp - worked out what is happening.
Nothing to do with extensions - I could reproduce with all extensions disabled.
@vasily-kirichenko - nothing to do with lenses. Lenses do work (eventually) and are really nice!

My solution has 22 projects, 21 C#, 1 F#. I have an extension called 'clean solution' which gives me a nice clean environment every time I start up. When I start up with no files open, everything initialises then calms down. 

When I then open an F# file, I can see background tasks and live code analysis, but even when that finishes, no intellisense in the file. I also see the message about 'Rechecking code due to buffer edit!' in the F# language service log.

If I then leave that file open, restart visual studio with that file open, background tasks run, it does live code analysis and I can even see the filename of the file it is analysing.  I see about 10 seconds worth of 'Rechecking code due to buffer edit!' messages, but once the analysis is complete everything is working, intellisense and breakpoints!

It works with and without full code solution analysis enabled.

So I actually thing this is a duplicate of #5170 ! It looks like the F# language service isn't initialised on a multi language solution unless an F# file is open on startup.

@cartermp - do you still want me to report a problem using VS, or do you have enough information?
